client/hwlib/src/helpers.rs

```diff
@@ -16,14 +16,30 @@
  *        Nadzeya Hutsko <nadzeya.hutsko@canonical.com>
  */
 
+use anyhow::{anyhow, Context, Result};
 use std::path::PathBuf;
+use time::Date;
 
 pub(crate) fn append_to_pathbuf(p: PathBuf, s: &str) -> PathBuf {
     let mut p = p.into_os_string();
     p.push(s);
     p.into()
 }
 
+pub(crate) fn normalize_date(date_str: &str) -> Result<Date> {
+    let parts: Vec<_> = date_str.trim().split('/').collect();
+    if parts.len() != 3 {
+        return Err(anyhow!("invalid date format: {date_str}"));
```

Comment:
This is very pedantic, but I think this error occurs if `date_str` is invalidly formatted, rather than `date_str` itself being an invalid format (`date_str != "mm-dd-yyyy"`)

Reply:
yes, I've just checked and it fails with  `cannot parse date "2-2-2022"`

Reply:
Ah, thanks for the context !